_posts/2024-08-25-mot-tracking-metrics.md

```diff
@@ -0,0 +1,172 @@
+2. It balances **precision and recall** of identity predictions.
+3. It's less affected by the number of objects in the scene than MOTA.
+
+However, IDF1 also has limitations:
+
+1. **IDF1 can decrease when improving detection**. Just avoiding FP can result in a better metric (A vs C)
+2. **IoU is fixed** so more or less detection accuracy is not reflected on the metric
+
+More limitations are presented in the HOTA paper. I recommend you to have a read because it is very well explained and intuitive.
+
+## HOTA (Higher Order Tracking Accuracy)
+
+HOTA is a more recent metric designed to address the limitations of both MOTA and IDF1. It aims to provide a balanced assessment of detection and association performance. HOTA can be broken down into DetA (Detection Accuracy) and AssA (Association Accuracy), allowing separate analyses of these aspects.
+
+The core HOTA formula is:
+
+$$
+ \text{HOTA}_{\alpha} = \sqrt{\text{DetA}_{\alpha} \cdot \text{AssA}_{\alpha}} 
+$$
+
+In this formula, the alpha term represents the different Intersection over Union (IoU) thresholds used to compute the metric. A True Positive (TP) is only considered when the match IoU score is above the given alpha threshold. The metric uses 19 different alpha values, ranging from 0.05 to 0.95 in increments of 0.05.
+
+HOTA uses global alignment (high-order association) between predicted and ground truth detections, similar to IDF1, but also incorporates localization accuracy. This means that HOTA evaluates both the ability to detect objects accurately and to maintain correct associations over time.
+
+The HOTA algorithm can be summarized in the following steps:
+
+```yaml
+For each frame
+	For each alpha value
+		Perform matching between gt and preds (Hungarian algorithm)
+		Obtain TP, FP and FN from previous matching
+		Compute association accuracy across the entire video for each TP.
+```
+
+<div class="post-center-image">
+    {% picture pimage /assets/images/fullsize/posts/2024-08-25-mot-tracking-metrics/hota.jpg --alt HOTA metric diagram %}
+</div>
+
+{:refdef: class="image-caption"}
+*Figure 4. HOTA metric diagram*
+{: refdef}
+
+HOTA drawbacks:
+
+- **Not Ideal for Online Tracking**: HOTA's association score depends on future associations across the entire video, making it less suitable for evaluating online tracking where future data isn't available.
+- **Doesn't Account for Fragmentation**: HOTA does not penalize fragmented tracking results, as it is designed to focus on long-term global tracking, which may not align with all application needs.
+
+If you want to learn more about HOTA, I recommend reading the blog post by [Jonathon Luiten](https://jonathonluiten.medium.com/how-to-evaluate-tracking-with-the-hota-metrics-754036d183e1){:target="_blank" rel="noopener noreferrer"}. Jonathon Luiten is one of the authors of the HOTA paper, and his post is an excellent resource for learning how to use the metric to compare different trackers.
+
```

Comment:
I think adding a section like "Part 3: How does HOTA compare to alternative tracking metrics." in the other post would be very helpful to visually understand the metrics and what they capture